examples/apps/kme/workload/kme_workload_plug-in.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+#include <map>
+#include "kme_workload_plug-in.h"
+
+REGISTER_WORKLOAD_PROCESSOR_EXT_WORK_ORDER("kme",KEY_MANAGEMENT_ENCLAVE,KMEWorkloadProcessor);
+
+void KMEWorkloadProcessor::GetUniqueId(
+    const std::vector<tcf::WorkOrderData>& in_work_order_data,
+    std::vector<tcf::WorkOrderData>& out_work_order_data,
+    ExtWorkOrderInfoKME* ext_wo_info_kme) {
+
+    ByteArray signing_key = {};
+    ByteArray verification_key_hex = {};
+    ByteArray verification_key_signature_hex = {};
+    ByteArray nonce_hex = in_work_order_data[0].decrypted_data;
+    
+    int err = ext_wo_info_kme->GenerateSigningKey(
+        ExtWorkOrderInfo::KeyType_SECP256K1, nonce_hex, signing_key,
+        verification_key_hex, verification_key_signature_hex);
+
+    SetStatus(err, out_work_order_data);
+    
+    if (!err) {
+        sig_key_map[verification_key_hex] = signing_key;
+        AddOutput(1, verification_key_hex, out_work_order_data);
+        AddOutput(2, verification_key_signature_hex, out_work_order_data);
+    }
+}
+
+void KMEWorkloadProcessor::Register(
+    const std::vector<tcf::WorkOrderData>& in_work_order_data,
+    std::vector<tcf::WorkOrderData>& out_work_order_data,
+    ExtWorkOrderInfoKME* ext_wo_info_kme) {
+
+    // To be implemented
+}  // KMEWorkloadProcessor::Register
+
+void KMEWorkloadProcessor::PreprocessWorkorder(
+    const std::vector<tcf::WorkOrderData>& in_work_order_data,
+    std::vector<tcf::WorkOrderData>& out_work_order_data,
+    ExtWorkOrderInfoKME* ext_wo_info_kme) {
+
+    // To be implemented
+}  // KMEWorkloadProcessor::PreprocessWorkorder
+
+void KMEWorkloadProcessor::AddOutput(int index, ByteArray& data,
```

Comment:
Add function comments what this function will do.

Reply:
Done